api/src/main/java/org/openmrs/module/queue/model/EncounterStatus.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.util.Date;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.openmrs.BaseOpenmrsMetadata;
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+
+@EqualsAndHashCode(callSuper = true)
+@NoArgsConstructor
+@Data
+@Entity
+@Table(name = "encounter_status")
+public class EncounterStatus extends BaseOpenmrsMetadata {
+
+	private static final long serialVersionUID = 1L;
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "encounter_status_id")
+	private Integer encounterStatusId;
+
+	@Column(name = "patient_id")
+	private Patient patient;
+
+	@Column(name = "encounter_type_id")
+	private EncounterType encounterType;
+
+	@Column(name = "encounter_id")
```

Comment:
Is there a possibility of this encounter having a different encounter type and encounter date time from those in the `encounterType` and `encounterDatetime` fields?

Reply:
@dkayiwa I think the idea here is that an encounter status might exist before an actual encounter has started. So we can use the `encounterType` field to track what sort of encounter *should* be created. Then, once the encounter actually exists, we reference the encounter for that information.

Reply:
Oh i see! Thanks for the explanation. 😊